tests/test_functions.py

```
@@ -0,0 +1,188 @@
+    def test_slopeofsomething_point(self):
+        sspeed_1 = np.array([1531.3815, 1531.0292, 1527.5515, 1524.2278, 1523.1734, 1522.6060, 1522.1262, 1521.7904, 1521.206, 1520.5105, 1519.8733, 1517.8005, 1514.4107, 1510.7368, 1507.0292, 1503.8638, 1500.0133, 1498.6095, 1498.5553, 1497.4988, 1497.2212])
+        depth_1 = np.array([55.7643, 65.8073, 77.8539, 92.3261, 109.729, 130.666, 155.851, 186.26, 222.475, 266.04, 318.127, 380.213, 453.938, 541.089, 643.567, 763.333, 902.339, 1062.44, 1245.29, 1452.25, 1684.28])
+        self.assertEqual(len(sspeed_1), len(depth_1))
+        test_result_1 = slopeofsomething_point(sspeed_1, depth_1)
+        self.assertAlmostEqual(test_result_1, -23.508440422965638)
+
+        sspeed_2 = np.array([1494.3858, 1494.3122, 1492.9553, 1493.0089, 1492.2201, 1487.9422, 1482.8552, 1481.0305])
+        depth_2 = np.array([55.7643, 65.8073, 77.8539, 92.3261, 109.729, 130.666, 155.851, 186.26])
+        
+        self.assertEqual(len(sspeed_2), len(depth_2))
+        test_result_2 = slopeofsomething_point(sspeed_2, depth_2)
```

Ruff (0.12.2)

`slopeofsomething_point` may be undefined, or defined from star imports

(F405)